### ISSUE TITLE
FilteredComposerIterator doesn't accept "inside" of psr-4 directory

### ISSUE THREAD

**Issue by mpyw** [alekitto/class-finder#3]
# Problem

Assume that there is `composer.json` of Laravel project.

```json
{
  "autoload": {
    "psr-4": {
      "App\\": "app/"
    }
  }
}
```

We need to find Eloquent `Model` classes inside of `app/Models`.

```php
$finder = (new ComposerFinder())
    ->subclassOf(Model::class)
    ->in(base_path('app/Models'));
```

But it can't find any files.

# Reason

Look at this class:

[class-finder/FilteredComposerIterator.php at 60ba72245f9ae5eb0d670d49e3457150c47d19bc · alekitto/class-finder](https://github.com/alekitto/class-finder/blob/60ba72245f9ae5eb0d670d49e3457150c47d19bc/lib/Iterator/FilteredComposerIterator.php)

If I fix the following line, it works as expected.

```diff
    private function validDir($path): bool
    {
        if (null === $this->dirs) {
            return true;
        }

        $path = PathNormalizer::resolvePath($path);
        foreach ($this->dirs as $dir) {
-           if (0 === \strpos($path, $dir)) {
+           if (0 === \strpos($dir, $path)) {
                return true;
            }
        }

        return false;
    }
```

Note that there are more similar statements in other filters that need to be fixed.

**Comment by alekitto** [comment]
Should be fixed in https://github.com/alekitto/class-finder/commit/ecbb7153814d81dccf3f29f78835ad8db45f17c9.
Could you please test the `master` version and confirm?

**Comment by mpyw** [comment]
@alekitto LGTM! Please release it!

**Comment by alekitto** [comment]
Fix released in 0.1.5.
